board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c

```diff
@@ -13,11 +13,50 @@
 #include <asm/arch/iomux.h>
 #include <asm/mach-imx/boot_mode.h>
 #include <asm/gpio.h>
+#include <fs.h>
 #include <usb.h>
 #include <dm.h>
+#include <spi_flash.h>
+#include <fpga.h>
+#include <mtd.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
+#define M4_BASE			(TCML_BASE)
+#define M4_SIZE			(SZ_128K + SZ_64K)
+#define M4_ENTRY_OFFSET		0x1004
+#define M4_WATERMARK_OFFSET	0x1000
+#define M4_WATERMARK(fw)						\
```

Comment:
Looks like most pieces here are 7ulp generic, might be good to move to the common files and just call the right functions here.

Reply:
done